### PR TITLE
Alternative to #5219 :super snowflake edition

### PR DESCRIPTION
**Pull request by Aurorablade**
Yes yes i know..snowflake species checks..

:cl: Fethas
tweak:Plasmaman can no longer be vampires
tweak:removes the mask check on both victim AND Vampire and makes the message a bit more obscure...so its up to your imagination of where you are biting them and with what..
bugfix: (Maybe?)..We actually check if the race HAS BLOOD before trying to suck them
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 1]
```diff
@@ -110,12 +110,20 @@
 		if(I_HARM)
 			//Vampire code
 			if(M.mind && M.mind.vampire && (M.mind in ticker.mode.vampires) && !M.mind.vampire.draining && M.zone_sel && M.zone_sel.selecting == "head" && src != M)
+				if(species && species.flags & NO_BLOOD)//why this hell were we never checkinf for this?
+					to_chat(M, "<span class='warning'>They have no blood!</span>")
```
Comment (Crazylemon64) [inline code/modules/mob/living/carbon/human/human_attackhand.dm]: Needs a return

Comment (Aurorablade) [inline code/modules/mob/living/carbon/human/human_attackhand.dm]: ....damn you're quick..

**Comment by Crazylemon64** [comment]
This PR sucks

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 2]
```diff
@@ -110,12 +110,20 @@
...
 			if(M.mind && M.mind.vampire && (M.mind in ticker.mode.vampires) && !M.mind.vampire.draining && M.zone_sel && M.zone_sel.selecting == "head" && src != M)
+				if(species && species.flags & NO_BLOOD)//why this hell were we never checkinf for this?
+					to_chat(M, "<span class='warning'>They have no blood!</span>")
 				if((head && (head.flags & HEADCOVERSMOUTH)) || (wear_mask && (wear_mask.flags & MASKCOVERSMOUTH)))
 					to_chat(M, "<span class='warning'>Remove their mask!</span>")
 					return
 				if((M.head && (M.head.flags & HEADCOVERSMOUTH)) || (M.wear_mask && (M.wear_mask.flags & MASKCOVERSMOUTH)))
-					to_chat(M, "<span class='warning'>Remove your mask!</span>")
-					return
+					if(M.get_species() == "Vox")
```
Comment (FlattestGuitar) [inline code/modules/mob/living/carbon/human/human_attackhand.dm]: This way of coding it makes it skip all other checks. Vox can suck skeletons dry. 😏

**Comment by Aurorablade** [comment]
i am seriously ope to people telling me what code to put where here....its like.2 am and i am coing...

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 3]
```diff
@@ -110,12 +110,20 @@
...
+						M.mind.vampire.handle_bloodsucking(src)
+						add_logs(src, M, "vampirebit")
+						msg_admin_attack("[key_name_admin(M)] vampirebit [key_name_admin(src)]")
+						return
```
Comment (Crazylemon64) [inline code/modules/mob/living/carbon/human/human_attackhand.dm]: You don't need to copy the entire bloodsucking section - just have it skip the "remove your mask" return, and the other checks will hold up fine

File: code/modules/mob/living/carbon/human/human_attackhand.dm (hunk 2 above)
Comment (FlattestGuitar) [inline code/modules/mob/living/carbon/human/human_attackhand.dm]: Make the mask check pass if they're a Vox. No need to double up the sucking code. 😏

File: code/game/gamemodes/vampire/vampire.dm [hunk 4]
```diff
@@ -273,7 +273,10 @@ You are weak to holy things and starlight. Don't go into space and avoid the Cha
 	owner.attack_log += text("\[[time_stamp()]\] <font color='red'>Bit [H] ([H.ckey]) in the neck and draining their blood</font>")
 	H.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has been bit in the neck by [owner] ([owner.ckey])</font>")
 	log_attack("[owner] ([owner.ckey]) bit [H] ([H.ckey]) in the neck")
-	owner.visible_message("<span class='danger'>[owner] bites [H]'s neck!</span>", "<span class='danger'>You bite [H]'s neck and begin to drain their blood.</span>", "<span class='notice'>You hear a soft puncture and a wet sucking noise.</span>")
+	if(owner.get_species() == "Vox")
```
Comment (TheDZD) [inline code/game/gamemodes/vampire/vampire.dm]: :hocho:

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 5]
```diff
@@ -110,12 +110,16 @@
...
+				if(species && species.flags & NO_BLOOD)//why this hell were we never checkinf for this?
+					to_chat(M, "<span class='warning'>They have no blood!</span>")
+					return
 				if((head && (head.flags & HEADCOVERSMOUTH)) || (wear_mask && (wear_mask.flags & MASKCOVERSMOUTH)))
 					to_chat(M, "<span class='warning'>Remove their mask!</span>")
 					return
 				if((M.head && (M.head.flags & HEADCOVERSMOUTH)) || (M.wear_mask && (M.wear_mask.flags & MASKCOVERSMOUTH)))
-					to_chat(M, "<span class='warning'>Remove your mask!</span>")
-					return
+					if(M.get_species() != "Vox")
```
Comment (TheDZD) [inline code/modules/mob/living/carbon/human/human_attackhand.dm]: :hocho: :hocho:

**Comment by TheDZD** [comment]
Honestly it might just be better to remove the whole thing with breath masks, one on hand it doesn't make a whole lot of sense, on the other, they're fucking bluespace vampires.

**Comment by Lady-Luck** [comment]
>Honestly it might just be better to remove the whole thing with breath masks, one on hand it doesn't make a whole lot of sense, on the other, they're fucking bluespace vampires.

Why can't a vampire drain the blood of someone wearing a breath mask, for that matter?

**Comment by Aurorablade** [comment]
@TheDZD i can if theres enough push for it, but i see it as thier fangs having to pierce the material of the mask...which kinda negates its usefulness as a oxygen mask...

**Comment by Fox-McCloud** [comment]
Inclined to agree with DZD, TBH.

**Comment by Lady-Luck** [comment]
Vampires shouldn't even need to breathe at all. Vox Vampires should just be trying to maintain the illusion they aren't a vampire.

**Comment by Fox-McCloud** [comment]
>Vampires shouldn't even need to breathe at all.

Way to easy to meta-test.

**Comment by Lady-Luck** [comment]
>Way to easy to meta-test.

In what sense? Removing a vox's breath mask on a hunch is a good way to get BWOINKed.

IT's not like you can't just drag them to the chapel.

**Comment by TheDZD** [comment]
@Lady-Luck Grab a health analyzer, and a syringe of charcoal, take off their mask, if they start suffocating, put it back on and inject them to fix the toxins.

**Comment by Belario** [comment]
That sort of "test" sounds like the sort of thing people should be getting job bans for.

**Comment by Fox-McCloud** [comment]
@Lady-Luck 

Super easy; just have mutadone and strangle them; it wouldn't even need to be a full out strangle; just a neck grab and you'd be able to tell.

**Comment by Crazylemon64** [comment]
Isn't randomly strangling people kind of a not-do-all-the-time thing

**Comment by Spacemanspark** [comment]
Yes, but I can still see security lining up everyone in the chapel or whatever to try and do another validhunt test like they did with the changeling blood + welding fuel shit.

**Comment by Lady-Luck** [comment]
>Super easy; just have mutadone and strangle them; it wouldn't even need to be a full out strangle; just a neck grab and you'd be able to tell.

Who in their right mind would do that instead of taking them to the chapel?

**Comment by Dinarzad** [comment]
Strangling someone because "They might be a Vamp" is sort of the same tier as killing random criminals cuz "He might be a traitor" and then incinerating the body because "He may be a Traitor-Ling."

There's always going to be a "Meta" test, but if security is fucking -strangling- people, that's a PRETTY open an shut case of Assault if they end up not being a Vamp, and a great way to be fired or Ahelped on.
(Side note, I don't have any opinion one way or another on vamps having no-breathe, but I don't feel this being cited as a meta test is viable given just how blatantly bwoinkable/IA territory it is.)

**Comment by Aurorablade** [comment]
removes the mask check on both victim AND Vampire and makes the message a bit more obscure...so its up to your imagination of where you are biting them and with what..

**Pull request merged**